metricbeat/module/mysql/query/query.go

```diff
@@ -0,0 +1,139 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/*
+Package status fetches MySQL server status metrics.
+
+For more information on the query it uses, see:
+http://dev.mysql.com/doc/refman/5.7/en/show-status.html
+*/
+package query
+
+import (
+	"context"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/metricbeat/helper/sql"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+func init() {
+	mb.Registry.MustAddMetricSet("mysql", "query", New,
+		mb.DefaultMetricSet(),
+	)
+}
+
+type query struct {
+	Namespace      string `config:"query_namespace" validate:"nonzero,required"`
+	Query          string `config:"query" validate:"nonzero,required"`
+	ResponseFormat string `config:"response_format" validate:"nonzero,required"`
+	DeDotEnabled   bool   `config:"dedot.enabled"`
```

Comment:
We are dedotting by default everywhere else, we should set it to true by default implementing `InitDefaults()`.
Also I am not sure if we should use the `dedot` name here, as we are using it to do the oposite, adding dots where there weren't before.

Reply:
Good catch, I took the name from `http.json` metricset but I read its behavior in reverse. Any suggestions for the name?